modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -1204,12 +1232,32 @@ public void onFailure(Exception e) {
      * is different to what's already in the error store.
      * This allows us to not spam the logs and only log new errors when we're about to record them in the store.
      */
-    static void recordAndLogError(String targetIndex, DataStreamLifecycleErrorStore errorStore, Exception e, String logMessage) {
-        String previousError = errorStore.recordError(targetIndex, e);
-        if (previousError == null || previousError.equals(errorStore.getError(targetIndex)) == false) {
+    static void recordAndLogError(
+        String targetIndex,
+        DataStreamLifecycleErrorStore errorStore,
+        Exception e,
+        String logMessage,
+        int signallingErrorRetryThreshold
+    ) {
+        ErrorEntry previousError = errorStore.recordError(targetIndex, e);
+        ErrorEntry currentError = errorStore.getError(targetIndex);
+        if (previousError == null || (currentError != null && previousError.error().equals(currentError.error()) == false)) {
             logger.error(logMessage, e);
```

Comment:
(academic question, doesn't have to be answered in this PR), do you think it would be useful to log what the previous error was and how often it occurred as we're replacing it with a new error? Otherwise something could log once, but  fail 8 more times, then be replaced with a new error, and we'd never know from the logs that it failed those other 8 times. What do you think?

Reply:
That's a very good question. I don't know. If it's made progress and self-healed or someone noticed the initial logging and fixed it manually, does it still matter? I'm tempted to say it doesn't anymore and we should keep the logging logic as simple as possible (we can probably find a way to make the logging UX nice, but having to parse two printed errors when only one is still relevant might be counter productive). Happy to add it if you think it brings value though.